ports/mimxrt10xx/supervisor/serial.c

```diff
@@ -26,27 +26,29 @@
  * THE SOFTWARE.
  */
 
+#include "supervisor/serial.h"
 #include "py/mphal.h"
 #include <string.h>
-#include "supervisor/serial.h"
 
 #include "fsl_clock.h"
 #include "fsl_lpuart.h"
 
 // TODO: Switch this to using DEBUG_UART.
-
+// Need official way to turn off the port serial code when it is not needed
+#if defined(USE_DEBUG_PORT_CODE)
```

Comment:
Note the DEBUG_UART pin definitions in `mimxrt10xx/boards/imxrt1060_evk/mpconfigboard.h`. These can be commented out if you don't want a debug UART. That is a way to turn the DEBUG UART on and off.

So then you can use `#if defined (CIRCUITPY_DEBUG_UART_RX)` as the guard instead of a new preprocessor macro name.

Reply:
Sounds good, I uploaded change that checked for either the RX or TX pin.
I verified that if I build for imxrt1060_evk the code is built in and if I build for example: teensy40 which does
not have these defines, it does not include this code and UART4 works fine and all of the logical serial ports
on the Teensy can talk to each other.